tests/test_metrics.py

```diff
@@ -190,6 +195,114 @@ def testValidEmpty(self):
+
+        result = task.run(self.startTask.getFullMetadata(), self.startTask.getFullMetadata())
+        oracle = altTask.run(self.startTask.getFullMetadata())
+
+        self.assertEqual(result.measurement.metric_name, self.metric)
+        self.assertAlmostEqual(result.measurement.quantity.to_value(u.s),
+                               oracle.measurement.quantity.to_value(u.s))
+
+    def testTwoTasks(self):
+        firstTask = PipelineTimingMetricTask(config=self._makeConfig(nameStart="first", nameEnd="first"))
+        secondTask = PipelineTimingMetricTask(config=self._makeConfig(nameStart="last", nameEnd="last"))
+
+        result = self.task.run(self.startTask.getFullMetadata(), self.endTask.getFullMetadata())
+        firstResult = firstTask.run(self.startTask.getFullMetadata(), self.startTask.getFullMetadata())
+        secondResult = secondTask.run(self.endTask.getFullMetadata(), self.endTask.getFullMetadata())
+
+        self.assertEqual(result.measurement.metric_name, self.metric)
+        self.assertGreater(result.measurement.quantity, 0.0 * u.s)
+        self.assertGreaterEqual(result.measurement.quantity,
+                                firstResult.measurement.quantity + secondResult.measurement.quantity)
+
+    def testRunDifferentMethodFirst(self):
+        config = self._makeConfig(nameStart="first", nameEnd="last")
+        config.targetStart = "first.doProcess"
+        task = PipelineTimingMetricTask(config=config)
+        try:
+            result = task.run(self.startTask.getFullMetadata(), self.endTask.getFullMetadata())
+        except lsst.pipe.base.NoWorkFound:
+            # Correct behavior
+            pass
+        else:
+            # Alternative correct behavior
+            testUtils.assertValidOutput(task, result)
+            meas = result.measurement
+            self.assertIsNone(meas)
+
+    def testRunDifferentMethodLast(self):
+        config = self._makeConfig(nameStart="first", nameEnd="last")
+        config.targetStart = "last.doProcess"
+        task = PipelineTimingMetricTask(config=config)
+        try:
+            result = task.run(self.startTask.getFullMetadata(), self.endTask.getFullMetadata())
+        except lsst.pipe.base.NoWorkFound:
+            # Correct behavior
+            pass
+        else:
+            # Alternative correct behavior
+            testUtils.assertValidOutput(task, result)
+            meas = result.measurement
+            self.assertIsNone(meas)
+
+    def testBadlyTypedKeys(self):
+        metadata = self.endTask.getFullMetadata()
+        endKeys = [key
+                   for key in metadata.paramNames(topLevelOnly=False)
+                   if "EndUtc" in key]
+        for key in endKeys:
+            metadata[key] = 42
+
+        with self.assertRaises(lsst.verify.tasks.MetricComputationError):
+            self.task.run(self.startTask.getFullMetadata(), metadata)
+
+
 # Hack around unittest's hacky test setup system
 del MetricTaskTestCase
```

Comment:
I know it's outside ticket, but this wouldn't be necessary if `MetricTaskTestCase` wasn't derived from `lsst.utils.tests.TestCase`. The concrete classes that implement the tests would have to explicitly derive from that, but it would prevent unittest from thinking the ABC is a test.

Reply:
Yes, but it would cause other problems (like code in one class being written against fields or methods in a completely different class). This is the lesser evil.